src/services/item/plugins/etherpad/serviceConfig.ts

```diff
@@ -0,0 +1,49 @@
+import { singleton } from 'tsyringe';
+
+import {
+  ETHERPAD_API_KEY,
+  ETHERPAD_COOKIE_DOMAIN,
+  ETHERPAD_PUBLIC_URL,
+  ETHERPAD_URL,
+} from '../../../../utils/config';
+import { ETHERPAD_API_VERSION } from './constants';
+
+export const API_KEY_FORMAT = /^[a-f\d]{64}$/;
+
+@singleton()
+export class EtherPadServiceConfig {
+  private readonly _url: string;
+  private readonly _apiKey: string;
+  private readonly _publicUrl: string;
+  private readonly _cookieDomain: string;
+  private readonly _apiVersion: string;
+
+  public constructor() {
+    // URL and ApiKey are validated on read in the config file.
+    this._url = ETHERPAD_URL;
+    this._apiKey = ETHERPAD_API_KEY;
+    this._publicUrl = ETHERPAD_PUBLIC_URL ?? this._url;
+    this._cookieDomain = ETHERPAD_COOKIE_DOMAIN ?? new URL(this._publicUrl).hostname;
+    this._apiVersion = ETHERPAD_API_VERSION;
+  }
+
+  public get url() {
+    return this._url;
+  }
+
+  public get apiKey() {
+    return this._apiKey;
+  }
+
+  public get publicUrl() {
+    return this._publicUrl;
+  }
+
+  public get cookieDomain() {
+    return this._cookieDomain;
+  }
+
+  public get apiVersion() {
+    return this._apiVersion;
+  }
```

Comment:
I don't remember what we've talked about exactly yesterday, but you could simply have `public readonly attributes`

Reply:
Yeah you're right !